### PR TITLE
Breaking: getBalance modified + broadcasted event improvements

### PR DESCRIPTION
**Pull request by Alex-Werner**
### Issue being fixed or implemented
Describe in task DPW-841, we had a non-systematic issue with our Balance event. 
There is two things there, first `getBalance(unconfirmed=true)` was actually returned conf + unconf. 
And the name was confusing. Additionnaly, account.getBalance was calculating itself (accessing store data), making Storage unable to access for our fix (as we now use Storage.calculateBalance in our `current` value of our Balance-Changed event).

### What was done
- Fix : Wrong input consideration in broadcastTransaction.impactAffectedInputs.
- Fix : Sometimes wrong balance on spending transaction
- Fix : Wrong total balance on certain Balance events. 
- Moved : Calculation of balance in Storage
- Breaking : getBalance is now : `getUnconfirmedBalance`, `getConfirmedBalance`, `getTotalBalance`
- Breaking : Events `BALANCE_CHANGED` is not `CONFIRMED_BALANCE_CHANGED` for avoiding confusion. 

### Notes

As is it also misleading, account has now three method `getUnconfirmedBalance`, `getConfirmedBalance`, `getTotalBalance`. 
The method have quite similar API that `getBalance` with the difference of the first params (conf/unconf) being dropped. Default behaviour (with no params) stays the same.

CC: @seigler Sorry it took long time and requires UI changes too.

### CODE REVIEW

File: examples/wallet-dapi-fromHDPubKey.js [hunk 1]
```diff
@@ -13,8 +17,8 @@ const account = wallet.getAccount();
 
 
 const start = async () => {
-  console.log('Balance Conf', await account.getBalance(false, false));
-  console.log('Balance Unconf', await account.getBalance(true, false));
+  console.log('Balance Conf', await account.getConfirmedBalance(false, false));
+  console.log('Balance Unconf', await account.getUnconfirmedBalance(true, false));
```
Comment (seigler) [inline examples/wallet-dapi-fromHDPubKey.js]: These two should have only one parameter, `displayDuffs`.

File: examples/wallet-dapi-porto-update.js [hunk 2]
```diff
@@ -35,14 +46,14 @@ account.events.on(EVENTS.CREATED, (info) => {
 account.events.on(EVENTS.STARTED, (info) => {
   console.log('STARTED');
 });
-account.events.on(EVENTS.BALANCE_CHANGED, (info) => {
+account.events.on(EVENTS.CONFIRMED_BALANCE_CHANGED, (info) => {
   console.log('BALANCE_CHANGED');
```
Comment (seigler) [inline examples/wallet-dapi-porto-update.js]: `console.log('CONFIRMED_BALANCE_CHANGED');`

File: examples/wallet-dapi-protectcave.js [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+  transport,
 });
 
 const account = wallet.getAccount();
 const start = async () => {
-  console.log('Balance Conf', await account.getBalance(false, false));
-  console.log('Balance Unconf', await account.getBalance(true, false));
+  console.log('Balance Conf', await account.getConfirmedBalance(false, false));
+  console.log('Balance Unconf', await account.getUnconfirmedBalance(true, false));
   console.log('New Addr', await account.getUnusedAddress().address);
```
Comment (seigler) [inline examples/wallet-dapi-protectcave.js]: These two should have only one parameter, `displayDuffs`.

File: examples/wallet-dapi-protectcave.js [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
 };
 account.events.on(EVENTS.GENERATED_ADDRESS, (info) => { console.log('GENERATED_ADDRESS'); });
-account.events.on(EVENTS.BALANCE_CHANGED, (info) => { console.log('Balance Changed', info, info.delta); });
+account.events.on(EVENTS.CONFIRMED_BALANCE_CHANGED, (info) => { console.log('Balance Changed', info, info.delta); });
```
Comment (seigler) [inline examples/wallet-dapi-protectcave.js]: 'Confirmed balance changed' or 'CONFIRMED_BALANCE_CHANGED'

File: examples/wallet-insight-smartrug.js [hunk 5]
```diff
@@ -9,11 +9,11 @@ const wallet = new Wallet({
 
 const account = wallet.getAccount();
 const start = async () => {
-  console.log('Balance Conf', await account.getBalance(false, false));
-  console.log('Balance Unconf', await account.getBalance(true, false));
+  console.log('Balance Conf', await account.getConfirmedBalance(false, false));
+  console.log('Balance Unconf', await account.getUnconfirmedBalance(true, false));
   console.log('New Addr', await account.getUnusedAddress());
```
Comment (seigler) [inline examples/wallet-insight-smartrug.js]: one parameter

File: examples/wallet-insight-smartrug.js [hunk 6]
```diff
@@ -9,11 +9,11 @@ const wallet = new Wallet({
...
 };
-account.events.on(EVENTS.BALANCE_CHANGED, (info) => { console.log('Balance Changed', info, info.delta); });
+account.events.on(EVENTS.CONFIRMED_BALANCE_CHANGED, (info) => { console.log('Balance Changed', info, info.delta); });
```
Comment (seigler) [inline examples/wallet-insight-smartrug.js]: updated string

File: examples/wallet-dapi-fromHDPubKey.js (hunk 1 above)
Comment (Alex-Werner) [reply examples/wallet-dapi-fromHDPubKey.js]: Thanks ! Fixed.

File: examples/wallet-dapi-protectcave.js (hunk 4 above)
Comment (Alex-Werner) [reply examples/wallet-dapi-protectcave.js]: Second one for consistency :)

**Pull request merged**